radixdlt-engine/src/main/java/com/radixdlt/engine/RadixEngine.java

```diff
@@ -412,17 +417,15 @@ private REProcessedTxn verify(EngineStore.EngineStoreInTransaction<M> engineStor
 		}
 
 		try {
-			batchVerifier.testMetadata(meta, processedTxns);
+			final var processedMetadata = batchVerifier.processMetadata(meta, engineStore, processedTxns);
```

Comment:
Need to expose an engineStore which has the dbTxn. Otherwise may get inconsistent results.